src/glue/jobs/s3_to_json.py

```diff
@@ -336,35 +344,89 @@ def write_file_to_json_dataset(
         s3_metadata["start_date"] = metadata["start_date"].isoformat()
     s3_metadata["end_date"] = metadata["end_date"].isoformat()
-    output_filename = get_output_filename(metadata=metadata)
-    output_path = os.path.join(dataset_identifier, output_filename)
-    data = []
+    part_number = 0
+    output_path = get_part_path(
+            metadata=metadata,
+            part_number=part_number,
+            dataset_identifier=dataset_identifier,
+            touch=True
+    )
     with z.open(json_path, "r") as input_json:
-        with open(output_path, "w+") as f_out:
-            for transformed_block in transform_block(
-                    input_json=input_json,
-                    dataset_identifier=dataset_identifier,
-                    metadata=metadata
-            ):
+        current_output_path = output_path
+        for transformed_block in transform_block(
+                input_json=input_json,
+                dataset_identifier=dataset_identifier,
+                metadata=metadata
+        ):
+            current_file_size = os.path.getsize(current_output_path)
+            if current_file_size > file_size_limit:
+                part_number += 1
+                print(f"!!! File is too large, creating new part {part_number}")
+                current_output_path = get_part_path(
+                        metadata=metadata,
+                        part_number=part_number,
+                        dataset_identifier=dataset_identifier,
+                        touch=True
+                )
+            with open(current_output_path, "a") as f_out:
                 for transformed_record in transformed_block:
                     f_out.write("{}\n".format(json.dumps(transformed_record)))
-    s3_output_key = os.path.join(
-        workflow_run_properties["namespace"],
-        workflow_run_properties["json_prefix"],
-        f"dataset={dataset_identifier}",
-        output_filename
-    )
-    logger.debug("Output Key: %s", s3_output_key)
-    with open(output_path, "rb") as f_in:
-        response = s3_client.put_object(
-                Body = f_in,
-                Bucket = workflow_run_properties["json_bucket"],
-                Key = s3_output_key,
-                Metadata = s3_metadata
+    uploaded_files = []
+    for part_file in os.listdir(dataset_identifier):
+        output_path = os.path.join(dataset_identifier, part_file)
+        s3_output_key = os.path.join(
+            workflow_run_properties["namespace"],
+            workflow_run_properties["json_prefix"],
+            f"dataset={dataset_identifier}",
+            part_file
+        )
+        logger.debug(
+                "Uploading %s to %s",
+                output_path,
+                s3_output_key
         )
-        logger.debug("S3 Put object response: %s", json.dumps(response))
-    if delete_upon_successful_upload:
-        os.remove(output_path)
+        with open(output_path, "rb") as f_in:
+            response = s3_client.put_object(
```

Comment:
The only side effect of using multipart uploads would be to enable us to have super large (>5 GB) NDJSON objects, which may give us issues later on in the JSON to Parquet job. There wouldn't be any speed advantages because boto can't upload parts in parallel unless I actually spawned multiple processes all running boto, which would come with its own complexities (I might [need to use an external library](https://stackoverflow.com/questions/72092993/i-want-to-use-boto3-in-async-function-python)?)